libs/vector_storage/vector_provider/mongodb.py

```diff
@@ -0,0 +1,107 @@
+from abc import ABC
+from typing import List, Optional
+
+from pymongo import MongoClient
+
+from libs.embedding.abstract import EmbeddingAbstract
+from libs.interfaces.document import Document, EmbeddingDocument, DocumentSearchFilters
+from libs.vector_storage.vector_provider.abstract import VectorProviderAbstract
+
+
+class MongoVectorProvider(VectorProviderAbstract, ABC):
+
+    def __init__(
+        self,
+        embedding_model: EmbeddingAbstract,
+        db_name: str,
+        mongodb_uri: str = "mongodb://localhost:27017/",
+    ):
+
+        self.client = MongoClient(mongodb_uri)
+        self.db = self.client[db_name]
+        self.document_collection = self.db["documents"]
+        super().__init__(embedding_model)
+
+    def search(
+        self,
+        query: str,
+        filters: Optional[DocumentSearchFilters] = None,
+        k: int = 100,
+        threshold: float = 0.9,
+    ) -> List[Document]:
+        if not filters:
+            filters = DocumentSearchFilters()
+
+        builded_filters = []
+
+        if filters.city:
+            if isinstance(filters.city, list):
+                builded_filters.append({"metadata.city": {"$in": filters.city}})
+            else:
+                builded_filters.append({"metadata.city": filters.city})
+
+        if filters.state:
+            if isinstance(filters.state, list):
+                builded_filters.append({"metadata.state": {"$in": filters.state}})
+            else:
+                builded_filters.append({"metadata.state": filters.state})
+
+        query_vector = self.embedding_model.encode(
+            query
+        )  # Encode the query string to a vector
+
+        if len(query_vector) == 0:
+            return []
+
+        results = self.document_collection.aggregate(
+            [
+                {
+                    "$vectorSearch": {
+                        "index": "default_vector_index",
+                        "path": "values",
+                        "filter": (
+                            {"$and": builded_filters} if builded_filters else None
+                        ),
+                        "queryVector": query_vector,
+                        "numCandidates": k * 10,
```

Comment:
Why num of candidates is k * 10 ? Not just k .
Also why we need limit k ?

Reply:
![image](https://github.com/carenet-il/carenet/assets/87449439/fdba2fff-81bf-45a8-817d-c9942426605f)
https://www.mongodb.com/docs/atlas/atlas-vector-search/vector-search-stage/#mongodb-pipeline-pipe.-vectorSearch

The docs of the "vectorSearch" pipeline should answer those questions